pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -270,3 +271,84 @@ func (r *Revision) GetLastPinned() (time.Time, error) {
 func (rs *RevisionStatus) duck() *duckv1beta1.Status {
 	return &rs.Status
 }
+
+// PropagateDeploymentStatus takes the Deployment status and applies its values
+// to the Revision status.
+func (rs *RevisionStatus) PropagateDeploymentStatus(original *appsv1.DeploymentStatus) {
+	ds := transformDeploymentStatus(original)
+	cond := ds.getCondition(deploymentConditionReady)
+	if cond == nil {
+		return
+	}
+
+	switch cond.Status {
+	case corev1.ConditionUnknown:
+		revCondSet.Manage(rs).MarkUnknown(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	case corev1.ConditionTrue:
+		revCondSet.Manage(rs).MarkTrue(RevisionConditionResourcesAvailable)
+	case corev1.ConditionFalse:
+		revCondSet.Manage(rs).MarkFalse(RevisionConditionResourcesAvailable, cond.Reason, cond.Message)
+	}
+}
+
+var depCondSet = apis.NewLivingConditionSet(
+	deploymentConditionProgressing,
+	deploymentConditionReplicaSetReady,
+)
+
+const (
+	// deploymentConditionReady means the underlying deployment is ready.
+	deploymentConditionReady = apis.ConditionReady
+	// deploymentConditionReplicaSetready inverts the underlying deployment's
+	// ReplicaSetFailure condition.
+	deploymentConditionReplicaSetReady apis.ConditionType = "ReplicaSetReady"
+	// deploymentConditionProgressing reflects the underlying deployment's
+	// Progressing condition.
+	deploymentConditionProgressing apis.ConditionType = "Progressing"
+)
+
+type deploymentStatus struct {
+	duckv1beta1.Status `json:",inline"`
+}
+
+func (ds *deploymentStatus) getCondition(t apis.ConditionType) *apis.Condition {
+	return depCondSet.Manage(ds).GetCondition(t)
+}
+
+func (ds *deploymentStatus) initializeConditions() {
+	depCondSet.Manage(ds).InitializeConditions()
+	// The absence of this condition means no failure has occured.
+	depCondSet.Manage(ds).MarkTrue(deploymentConditionReplicaSetReady)
```

Comment:
You are not checking for the absence of this condition?

Reply:
Moved this around a bit to hopefully make it clearer what's going on. This will get overwritten by the deployment's condition if it's present, otherwise we assume it's true.